test/distributed/test_c10d_common.py

```diff
@@ -1532,6 +1532,18 @@ def tearDown(self):
         except OSError:
             pass
 
+    def test_init_process_group_optional_backend(self):
+        with tempfile.NamedTemporaryFile() as f:
+            store = dist.FileStore(f.name, self.world_size)
+            # creates both gloo and nccl backend
+            if dist.is_gloo_available() and dist.is_nccl_available():
+                dist.init_process_group(
+                    store=store,
+                    rank=self.rank,
+                    world_size=self.world_size,
+                )
+                dist.destroy_process_group()
+
```

Comment:
Would you like to add two all_reduce calls (one with cpu, the other with gpu) to make sure it is actually working? :)
If we already have such test, please ignore me :)
Also, it can be in a separate PR if we don't have it yet.

Reply:
Sounds good, it should work since we support gloo + nccl subgroups which would follow similar logic as the new backends. I will definitely refactor the existing tests for dispatchable collectives to include this though